### PR TITLE
Arc-enabled SQL Server - Defender for Cloud

### PR DESCRIPTION
**Pull request by chintalavr**
This PR includes scripts and documentation created for Arc-enabled SQL Server - Defender for Cloud. Please review and provide feedback for final changes.

### CODE REVIEW

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
+---
+type: docs
+title: "Arc-enabled SQL servers - Microsoft Defender for SQL servers on machines"
```
Comment (dkirby-ms) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:3]:
```suggestion
title: "Azure Arc-enabled SQL server - Microsoft Defender for SQL servers on machines"
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+linkTitle: "SQL servers on machines - Defender for Cloud"
+weight: 1
+---
+
+## Deploy an Azure Virtual Machine with Windows Server with Hyper-V to support nested virtualization
```
Comment (dkirby-ms) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:8]: this title should mention SQL and defender

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+The following Jumpstart scenario will walk you through on how to use the provided [Azure ARM Template](https://docs.microsoft.com/azure/azure-resource-manager/templates/overview) to deploy an Azure VM installed with Windows Server, setup Hyper-V to support nested virtualization, and create guest VM with SQL Server 2019 on Hyper-V to demonstrate Defender for Cloud for SQL servers on machines and generate alerts for SQL attacks.
+
+By the end of the guide, you will have an Azure VM **ArcBox-Client** installed with Windows Server 2019 with Hyper-V and nested Windows Server VM **ArcBox-SQL** pre-configured with SQL Server 2019, projected as an Azure Arc-enabled SQL Server, then enabled SQL assessment and Microsoft Defender for SQL servers on machines.
```
Comment (dkirby-ms) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:12]: Why is the VM named ArcBox-Client if this is not ArcBox?

Comment (dkirby-ms) [reply docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:12]: Same comment for Arcbox-SQL. We should avoid using ArcBox as a name unless this is an actual ArcBox flavor

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+As part of this scenario deployment Defender for SQL Servers on machine plan is enabled and Log Analytics workspace created in the scenario will be assigned as default workspace for Defender for Cloud.
```
Comment (dkirby-ms) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:14]: this sentence is a bit confusing, not sure what its trying to convey

**Review by dkirby-ms — CHANGES_REQUESTED** [review]
I see some ArcBox names in the documentation and in the screenshots. These should be changed to something more appropriate for a non-Arcbox scenario. Jumpstart-SQL or Jumpstart-Client, for example. 

Also provided some comments where I think we need more clarity.

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md (hunk 3 above)
Comment (chintalavr) [reply docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:12]: Using a nested VM in Hyper-V for this scenario and used ArcBox as it has built-in capabilities. Any suggestions for the name? I thought of changing name needed after review.

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md (hunk 4 above)
Comment (chintalavr) [reply docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:14]: It is just note to the user about the Defender for SQL plan is enabled as it has cost associated with it. It will enable for all machines in the subscription, even though we create one VM in this scenario. Let me rephrase little bit.

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md (hunk 3 above)
Comment (chintalavr) [reply docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:12]: Fixed all naming conventions

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
+---
+type: docs
+title: "Azure Arc-enabled SQL server - Microsoft Defender for SQL servers on machines"
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:3]:
```suggestion
title: "Azure Arc-enabled SQL Server and Microsoft Defender for SQL (Windows)"
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+linkTitle: "SQL servers on machines - Defender for Cloud"
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:4]:
```suggestion
linkTitle: "Azure Arc-enabled SQL Server and Microsoft Defender for SQL (Windows)"
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+weight: 1
+---
+
+## Deploy an Azure Virtual Machine with Windows Server with Hyper-V to support nested virtualization
+
+The following Jumpstart scenario will walk you through on how to use the provided [Azure ARM Template](https://docs.microsoft.com/azure/azure-resource-manager/templates/overview) to deploy an Azure VM installed with Windows Server, setup Hyper-V to support nested virtualization, and create guest VM with SQL Server 2019 on Hyper-V to demonstrate Defender for Cloud for SQL servers on machines and generate alerts for SQL attacks.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:10]:
```suggestion
The following Jumpstart scenario will walk you through how to use the provided [Azure ARM Template](https://docs.microsoft.com/azure/azure-resource-manager/templates/overview) to deploy an Azure VM installed with Windows Server, setup Hyper-V to support nested virtualization, and create guest VM with SQL Server 2019 on Hyper-V to demonstrate Defender for Cloud for SQL servers on machines and generate alerts for SQL attacks.
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+  - Microsoft.HybridCompute
+  - Microsoft.OperationsManagement
+  - Microsoft.HybridConnectivity
+  - Microsoft.GuestConfiguration
+  
+## Automation Flow
+
+The automation for this scenario includes different PowerShell scripts executed in the following order:
+
+1. [*Bootstrap.ps1*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_sqlsrv_jumpstart/azure/defendersql/scripts/Bootstrap.ps1) - Executed at ARM Template deployment time as a CustomScriptExtension. This script has the following functionalities:
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:79]:
```suggestion
1. [_Bootstrap.ps1_](https://github.com/microsoft/azure_arc/blob/main/azure_arc_sqlsrv_jumpstart/azure/defendersql/scripts/Bootstrap.ps1) - Executed at ARM Template deployment time as a CustomScriptExtension. This script has the following functionalities:
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+
+2. [*ArcServersLogonScript.ps1*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_sqlsrv_jumpstart/azure/defendersql/scripts/ArcServersLogonScript.ps1) - Executed upon initial login to the **JS-Client** Azure virtual machine. This script has the following functionalities:
+
+    1. Install Windows Hyper-V server and configure networking.
+    2. Create a guest Windows Server VM with SQL Server pre-installed.
+    3. Restore AdventureWorksLT2019 Database.
+    4. Execute the *ArcServersLogonScript.ps1* script.
+    5. Enable Defender for SQL Servers on Machine at subscription level and setup default Log Analytics workspace.
+    6. Execute the *testDefenderForSQL.ps1* script to simulate SQL attacks.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:91]:
```suggestion
    1. Install Windows Hyper-V server and configure networking.
    2. Create a guest Windows Server VM with SQL Server pre-installed.
    3. Restore _AdventureWorksLT2019_ Database.
    4. Execute the *ArcServersLogonScript.ps1* script.
    5. Enable Defender for SQL Servers on Machine at the subscription level and setup the default Log Analytics workspace.
    6. Execute the *testDefenderForSQL.ps1* script to simulate SQL attacks.
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+
+3. [*installArcAgentSQLSP.ps1*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_sqlsrv_jumpstart/azure/defendersql/scripts/installArcAgentSQLSP.ps1) - This is the main script and will be executed by the *ArcServersLogonScript.ps1* script at VM runtime. This script has the following functionalities:
+
+    1. Project SQL Server as an Azure Arc-enabled SQL server resource
+    2. Install Log Analytics agent using an extension on Azure Arc-enabled server
+    3. Create SQL Assessment and inject data to Azure Log Analytics workspace
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:97]:
```suggestion
    1. Project SQL Server as an Azure Arc-enabled SQL server resource
    2. Install the Log Analytics agent using an extension on the Azure Arc-enabled server
    3. Create SQL Assessment and inject data to Azure Log Analytics workspace
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+
+To get familiar with the automation and deployment flow read the following explanation.
+
+1. User edits the ARM template parameters file (1-time edit). These parameters values are used throughout the deployment.
+
+2. The ARM template includes an Azure VM Custom Script Extension which will deploy the [*Bootstrap.ps1*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_sqlsrv_jumpstart/azure/defendersql/scripts/Bootstrap.ps1) PowerShell Script. The script will:
+
+    1. Download the *ArcServersLogonScript.ps1*,  *installArcAgentSQLSP*, and  *testDefenderForSQL.ps1* PowerShell scripts
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:105]:
```suggestion
    1. Download the *ArcServersLogonScript.ps1*,  *installArcAgentSQLSP*, and *testDefenderForSQL.ps1* PowerShell scripts
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+
+  ![Screenshot showing Microsoft Defender for Cloud plans](./microsoft-defender-plans.png)
+
+  ![Screenshot showing Microsoft Defender for Cloud SQL enabled](./defender-sql-plan.png)
+
+- Following screen shows Arc-enabled SQL server Defender for Cloud enablement and protection status. Defender for Cloud for SQL Server is enabled at the subscription level, but protection status is still showing is not enabled. Please note it may take sometime to show this status in Azure portal, but still able to detect SQL threats generated by the test scripts.
+
+  ![Screenshot showing Microsoft Defender for Cloud - Arc-enabled SQL server status](./arcsql-defender-status.png)
+
+- Following screen shows the SQL threat detected by Defender for Cloud.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:250]:
```suggestion
- The below screenshot shows the SQL threat detected by Defender for Cloud.
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
+
+  ![Screenshot showing Defender for SQL security incidents and alerts](./defender-sql-security-incidents.png)
+
+- Following is the test script used to generate SQL threats, detect, and alert using Defender for Cloud for SQL servers.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:254]:
```suggestion
- The below screenshot shows the test script used to generate SQL threats, detect, and alert using Defender for Cloud for SQL servers.
```

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 14]
```diff
@@ -0,0 +1,269 @@
...
+
+  ![Screenshot showing Defender for SQL test scripts](./defender-sql-testing-script.png)
+
+- Please note once in a while these test execution may fails randomly. If you don't find these alerts, login to nested SQL VM in Hyper-V and execute test script manually as show below.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:258]: Why it is failing?

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 15]
```diff
@@ -0,0 +1,269 @@
...
+
+  ![Screenshot showing manual execution of the test scripts](./manual-brute-force-test.png)
+
+- Following is an email alert sent by Defender for Cloud when SQL threat is detected. By default this email is sent to the registered contact email at the subscription level.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:262]:
```suggestion
- The below screenshot shows an email alert sent by Defender for Cloud when a SQL threat is detected. By default, this email is sent to the registered contact email at the subscription level.
```

**Review by likamrat — CHANGES_REQUESTED** [review]
- Remove ArcBox references
- Fix Logs local path
- Remove unnecessary files
- Make sure _JS-Client_ is getting the right wallpaper
- Lingo and syntax
- Markdown standardization

File: azure_arc_sqlsrv_jumpstart/azure/defendersql/scripts/Bootstrap.ps1 [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+foreach ($app in $appsToInstall)
+{
+    Write-Host "Installing $app"
+    & choco install $app /y -Force | Write-Output
+}
+
+Write-Header "Fetching GitHub Artifacts"
+
+Write-Host "Fetching Artifacts for All Flavors"
+Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/main/img/arcbox_wallpaper.png" -OutFile $Env:ArcJSDir\wallpaper.png
```
[the remark this answers is not in the document]
Comment (chintalavr) [reply azure_arc_sqlsrv_jumpstart/azure/defendersql/scripts/Bootstrap.ps1:96]: Removed reference

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md (hunk 4 above)
Comment (chintalavr) [reply docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md]: Updated based on your comments

File: azure_arc_sqlsrv_jumpstart/azure/defendersql/scripts/LogInstructions.txt [hunk 17]
```diff
@@ -0,0 +1,7 @@
+To upload the log bundle zip file for support review, use the below command:
+
+azcopy copy "C:\ArcBox\Logs\<Logs Bundle zip file name>.zip" "https://https://jumpstartsupport.blob.core.windows.net/logbundles?sp=cw&st=2022-02-02T22:04:39Z&se=2027-04-22T05:04:39Z&spr=https&sv=2020-08-04&sr=c&sig=2urJGqXvkXTSBpu%2FqCksANy18pY5o44tmzzKjc8eUoM%3D" --recursive
+
+For example:
+
+azcopy copy "C:\ArcBox\Logs\LogsBundle-kavpug.zip" "https://https://jumpstartsupport.blob.core.windows.net/logbundles?sp=cw&st=2022-02-02T22:04:39Z&se=2027-04-22T05:04:39Z&spr=https&sv=2020-08-04&sr=c&sig=2urJGqXvkXTSBpu%2FqCksANy18pY5o44tmzzKjc8eUoM%3D" --recursive
```
[the remark this answers is not in the document]
Comment (chintalavr) [reply azure_arc_sqlsrv_jumpstart/azure/defendersql/scripts/LogInstructions.txt]: Updated as per your comments

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md [hunk 18]
```diff
@@ -0,0 +1,269 @@
...
+
+    2. Set local OS environment variables
+
+## Deployment
+
+As mentioned, this deployment will use an ARM Template. You will deploy a single template that creates all the Azure resources in a single resource group as well as onboarding the nested Hyper-V guest SQL Server VM to Azure Arc.
+
+- Before deploying the ARM template, login to Azure using AZ CLI with the ```az login``` command.
+
+- The deployment uses the ARM template parameters file. Before initiating the deployment, edit the [*azuredeploy.parameters.json*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_sqlsrv_jumpstart/azure/defendersql/azuredeploy.parameters.json) file located in your local cloned repository folder. An example parameters file is located [here](https://github.com/microsoft/azure_arc/blob/main/azure_arc_sqlsrv_jumpstart/azure/defendersql/azuredeploy.parameters.example.json).
```
[the remark this answers is not in the document]
Comment (chintalavr) [reply docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:115]: You are right missed this file. Created new one and committed.

File: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md (hunk 14 above)
Comment (chintalavr) [reply docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_defender/_index.md:259]: There are some bugs in the PS commands. This is what I have escalated to the engineering team review and fix. This is not a showstopper for now. New script is consistently generating alerts and good with the scenario.

**Review by likamrat — APPROVED** [review]
LGTM
